### ISSUE TITLE
The local_time and unix_time columns in the time table are misleading/incorrect

### ISSUE THREAD

**Issue by Smjert** [osquery/osquery#6743]
# Bug report

### What operating system and version are you using?
Ubuntu 18.04

### What version of osquery are you using?
osquery master (7627af388d3d66d9bc1d73a17596f490dc8ff1a1)

### What steps did you take to reproduce the issue?
1) `osqueryd -S`
2) `SELECT * from time`

### What did you expect to see?
```
osquery> select * from time;
       weekday = Saturday
          year = 2020
         month = 10
           day = 31
          hour = 12
       minutes = 28
       seconds = 41
      timezone = GMT
    local_time = 1604150921 <---- different
local_timezone = CET
     unix_time = 1604147321 <---- different
     timestamp = Sat Oct 31 12:28:41 2020 UTC
      datetime = 2020-10-31T12:28:41Z
      iso_8601 = 2020-10-31T12:28:41Z
```

### What did you see instead?
```
osquery> select * from time;
       weekday = Saturday
          year = 2020
         month = 10
           day = 31
          hour = 12
       minutes = 28
       seconds = 41
      timezone = GMT
    local_time = 1604147321 <---- same
local_timezone = CET
     unix_time = 1604147321 <---- same
     timestamp = Sat Oct 31 12:28:41 2020 UTC
      datetime = 2020-10-31T12:28:41Z
      iso_8601 = 2020-10-31T12:28:41Z
```

### Details
First of all, we have to take into account that `--utc=true` is the default; `--utc=true` says "--utc                                            Convert all UNIX times to UTC".
This is the first bit which is confusing/misleading. A Unix time is always expressed in UTC, there should be no conversion to be done.

Then we expect that `local_time` is different from `unix_time`, since I'm in CET (currently GMT+1), but it's not like that.
Looking at the description of both columns:
- local_time "Current local UNIX time in the system"
- unix_time "Current UNIX time in the system, converted to UTC if --utc enabled"

Starting from `local_time`, again a "local UNIX time" is not 100% clear. A UNIX time is UTC, maybe the description should say "Current time in seconds since Epoch, adjusted by the local timezone". It's not a UNIX time anymore, they are just seconds since the Epoch plus (or minus) the seconds of the timezone.

Then looking at the `unix_time` description, as we've already seen earlier, there should be no conversion needed, especially since the value for that column comes from `std::time()`.

On top of all this, even if we interpreted `local_time` with the corrected description, it still doesn't contain the correct value.
Even changing `--utc=false` doesn't matter, doing so only changes the `timezone` and the broken down date (the hour, minutes, seconds part).
Looking at the code the issue is here:
https://github.com/osquery/osquery/blob/7627af388d3d66d9bc1d73a17596f490dc8ff1a1/osquery/tables/utility/time.cpp#L43-L45
(The `local_time` variable is the result of calling `std::time()`).

The `std::mktime` function is taking a `tm` structure which considers to be in local time, and transforms it back to a `time_t` value in UTC.
The issue is that it doesn't keep the time unchanged, with the local timezone offset, but it adjusts it back to UTC using the local timezone by subtracting local timezone offset from the value. This is not the representation we want.

One way to fix this would be to trick `mktime` to think that the `tm` structure we pass instead of being in local time, it's actually in UTC already, therefore no adjustment will be made.
The problem with this is that taking the current timezone it's a bit long winded and setting a new one needs you to change environment variables.

An alternative is to use `timegm()`, which takes a `tm` structure, considers it being in UTC time, and converts it into a `time_t`.

### Summary:
1) `--utc=true` description is a bit confusing, UNIX time should not need no conversion, what it ends up changing for the `time` table is only the broken down representation of the date/time, which is not a UNIX time. Maybe it should just refer to "time" in general, but it's also difficult to understand which times is supposed to affect.
2) The description of the `unix_time` column is confusing, because a UNIX time should not need no conversion.
3) The description of the `local_time` column is confusing, again due to the mention of a "local UNIX time".
4) The `local_time` value is incorrect if we expect that to be a timestamp, which represent the seconds since the Epoch, adjusted by the timezone.

**Comment by theopolis** [comment]
Indeed, @Smjert do you mind addressing the confusion and fixing `local_time` such that it is adjusted?

**Comment by Smjert** [comment]
> Indeed, @Smjert do you mind addressing the confusion and fixing `local_time` such that it is adjusted?

Sure!
I think though I'm still a bit unclear on the `--utc` usage and the other descriptions, to understand how they should behave with the rest.
But maybe I can split this concerns on a different issue?

**Comment by theopolis** [comment]
Ah yes, perhaps we should deprecate the `--utc` flag (make it hidden and have it be a no-op). Then `local_time` is always the adjusted UNIX time.

**Comment by Smjert** [comment]
So, I temporarily closed the PR because it was failing on Windows, since I forgot that the table was available there too and `timegm` is not available on Windows.

Though this made me think again about the reason why we show a `local_time` UNIX time at all.
What real use does it have?
A UNIX timestamp should be sufficient to have an unambiguous point in time that's valid for all timezones.
That timestamp is fine for software and even humans to establish order of operations (if for instance used as a decoration to a query).

Finally if you want that time to be in your local time, I would say that you also want that to be really human readable and so I would expect one to transform it to a broken down view of day, month, year etc, which we already provide with the respective columns, when `--utc` is set to false.

TLDR: I'm actually proposing to remove/deprecate the `local_time` column, since it shows an incorrect time and it doesn't seem to add value to what we already show in the `time` table.

**Comment by theopolis** [comment]
That's a great plan!
